utils/sldgen/sld_generate.py

```diff
@@ -1542,19 +1561,28 @@ def gen_tile_acc(accelerator_list, axi_acceleratorlist, template_dir, out_dir):
     dp = dp_str.replace(acc + "_", "")
     dp_info = dp.split("_")
     skip = False
+    datatype = ""
     for item in dp_info:
       if re.match(r'dma[1-9]+', item, re.M|re.I):
         dp_dma_width = int(item.replace("dma", ""))
         if dp_dma_width != dma_width:
           skip = True
           break;
+      if re.match(r'flin', item, re.M|re.I):
+        datatype = "float_in"
+      elif re.match(r'flout', item, re.M|re.I):
+        datatype = "float_out"
+      elif re.match(r'fl', item, re.M|re.I):
```

Comment:
These regexp are easy to match with other words. We should define something similar to `dma[1-9]+` above.
E.g. fl32in, fl32out, fl32.
I understand this changes a bit the tcl script as well, but I really think it's best to make this match more reliable.

Reply:
Adding the 32 is a good idea. In addition, how about using `re.fullmatch` and removing the `re.I` flag? That would be that we are only matching full words enclosed by underscore characters: `_fl32_`, `_fl32out_`, `_fl32in_`.

Reply:
Good idea, agreed!

Reply:
Please make sure this works even when the 'fl32' is at the end of the HLS configuration string though.

Reply:
It does, I've tested that.